src/test/java/school/hei/haapi/integration/FeeIT.java

```diff
@@ -83,29 +95,61 @@ void teacher_read_ko() {
   @Test
   void manager_read_ok() throws ApiException {
     ApiClient manager1Client = anApiClient(MANAGER1_TOKEN);
-
     PayingApi api = new PayingApi(manager1Client);
-    Fee actual = api.getStudentFeeById(STUDENT1_ID, FEE1_ID);
 
-    assertEquals(fee1(), actual);
+    List<Fee> actual = api.getStudentFees(STUDENT1_ID);
+
+    assertTrue(actual.contains(fee1()));
   }
 
   @Test
-  void student_read_ko() {
+  void student1_read_student2_fees_ko() {
     ApiClient student1Client = anApiClient(STUDENT1_TOKEN);
-
     PayingApi api = new PayingApi(student1Client);
+
     assertThrowsApiException(
         "{\"type\":\"403 FORBIDDEN\",\"message\":\"Access is denied\"}",
-        () -> api.getStudentFeeById(STUDENT2_ID, FEE1_ID));
+        () -> api.getStudentFeeById(STUDENT2_ID, FEE2_ID));
   }
 
   @Test
   void student_read_own_ok() throws ApiException {
     ApiClient student1Client = anApiClient(STUDENT1_TOKEN);
+    PayingApi api = new PayingApi(student1Client);
+
+    List<Fee> actual = api.getStudentFees(STUDENT1_ID);
+
+    assertTrue(actual.contains(fee1()));
+  }
 
+  @Test
+  void manager_write_ok() throws ApiException {
+    ApiClient manager1Client = anApiClient(MANAGER1_TOKEN);
+    PayingApi api = new PayingApi(manager1Client);
+
+    List<Fee> newFees = api.createStudentFees(STUDENT1_ID, List.of(creatableFee1()));
+
+    List<Fee> actualFees = api.getStudentFees(STUDENT1_ID);
```

Comment:
actualFees --> expectedFees. And you have to manually define what is expected. Your test has no meaning if you build expected the same way you build actual (here, by reading db, as you db write might have failed and your test would not detect it since both expected and actual have equally failed)